core/src/main/scala/at/logic/gapt/proofs/ceres/StructCreators.scala

```
@@ -155,8 +172,45 @@ object StructCreators extends Logger {
         val cutAncInAntecedent = cutanc_seq.antecedent.map( x => Dual[Data]( A( x, Nil ) ) )
         val cutAncInSuccedent = cutanc_seq.succedent.map( x => A[Data]( x ) )
         val structs: Vector[Struct[Data]] = cutAncInAntecedent ++ cutAncInSuccedent
-        val r = Times[Data]( structs, List[Data]() )
-        r
+        if ( !proofLink.matches( "" ) ) {
+          //This code matches positiions for terms passed through the proof links
+          val ( Apps( _, vs ), hs: HOLSequent ) = ctx.get[ProofNames].names.get( proofLink ) match {
+            case Some( ( ex, hs ) ) => ( ex, hs )
+            case None               => ( Const( "", Ti ), HOLSequent() )
```

Just let the code fail in cases that should not happen.  There is no point in producing wrong results, an exception is always easier to debug.